src/ensembl/src/content/app/browser/track-panel/trackPanelActions.ts

```diff
@@ -82,6 +114,49 @@ export const changeTrackPanelModalViewForGenome: ActionCreator<
   );
 };
 
+export const updateBookmarksAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = () => (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const activeGenomeId = getBrowserActiveGenomeId(state);
+  const activeEnsObject = getBrowserActiveEnsObject(state);
+  if (!activeGenomeId || !activeEnsObject) {
+    return;
+  }
+  const trackStates = getBrowserTrackStates(state)[activeGenomeId];
+
+  const activeGenomeBookmarks = [...getActiveGenomeBookmarks(getState())];
```

Comment:
state is already extracted using getState above

Reply:
That's done